src/tye-diag-agent/Program.cs

```diff
@@ -30,17 +30,17 @@ public static void Main(string[] args)
                         var section = configuration.GetSection("provider");
                         foreach (var child in section.GetChildren())
                         {
-                            if (!DiagnosticsProvider.TryParse(child.Value, DiagnosticsProvider.ProviderKind.Unknown, out var provider))
-                            {
-                                throw new InvalidOperationException("Could not parse diagnostics provider: " + child.Value);
-                            }
-
                             // The value used to connect to the service should be accessible with Tye service discovery.
-                            if (provider.Value is string && provider.Value.StartsWith("service:"))
+                            DiagnosticsProvider provider;
+                            if (child.Value is string && child.Value.StartsWith("service:"))
```

Comment:
change here is that we were modeling providers in command line args as an array:

```sh
--provider:0=zipkin=...
--provider:1=elastic=...
```

Now they are an object:

```sh
--provider:zipkin=...
--provider:elastic=...
```

Much easier to generate and less arcane.